mono/utils/mono-threads.h

```diff
@@ -278,6 +278,9 @@ typedef struct _MonoThreadInfo {
 	 * TODO support multiple values by multiple tools
 	 */
 	void *tools_data;
+
+	/* Total bytes allocated by this thread in its lifetime so far. */
+	gint64 total_bytes_allocated;
```

Comment:
I'd rather have this variable on the SgenThreadInfo and access it through the gc interface (see for example mono_gc_get_used_size used for GC.GetTotalMemory). We will need special computation anyway in the GC, since we can't just bump this thread local variable for every allocation due to perf reasons.

Reply:
Is there some where that defines the gc interface? I see that all the gc's implement a mono_gc_get_used_size(void); but is there somewhere that specifies this is part of the interface?

Reply:
Not really. You can define the prototype in gc-internals.h and have all gc's (boehm, null) implement that method. That's it